aten/src/ATen/native/TensorTransformations.h

```diff
@@ -33,4 +33,21 @@ static inline void flip_check_errors(int64_t total_dims, int64_t flip_dims_size,
     ", but unique flip dims size=", flip_dims_v.size());
 }
 
+static inline Tensor roll_common(const Tensor& self, IntList shifts, IntList dims) {
+  AT_CHECK(shifts.size() > 0, "`shifts` required");
+  if (dims.size() == 0 && shifts.size() == 1) {
+    auto flattened = self.contiguous().view(self.numel());
+    return roll(flattened, shifts[0], 0).view(self.sizes());
+  }
+  AT_CHECK(
+    shifts.size() == dims.size(),
+    "shifts and dimensions must align. shifts: ", shifts.size(), ", dims:", dims.size()
+  );
+  AT_CHECK(dims.size() > 1, "this code should only be reached for handling multi-dim rolling" );
```

Comment:
nit: this should be AT_ASSERT, which is for internal errors.

Reply:
updated